### PR TITLE
Put lsp-use-plists in lsp-doctor

### PR DESCRIPTION
**Pull request by yyoncho**

### CODE REVIEW

**Review by ericdallo — APPROVED** [review]

File: lsp-mode.el [hunk 1]
```diff
@@ -8102,7 +8102,7 @@ This avoids overloading the server with many files when starting Emacs."
                                     (let ((res (with-current-buffer buf
                                                  ,form)))
                                       (cond
-                                       ((eq res :optional) (propertize "NOT AVAILABLE (OPTIONAL)" 'face 'warning))
+                                       ((eq res :optional) (propertize "OPTIONAL" 'face 'warning))
```
Comment (kiennq) [inline lsp-mode.el:8105]: Perhaps we should add `:not-available` option here? The `gccemacs` is not available in all Emacs release no

Comment (ericdallo) [reply lsp-mode.el:8105]: the gccemacs was merged into master, does it affect this check somehow?

Comment (kiennq) [reply lsp-mode.el:8105]: No, native compilation is still optional feature of Emacs

**Review by jcs090218 — APPROVED** [review]

**Comment by ericdallo** [comment]
Can we merge this @yyoncho ?

**Comment by yyoncho** [comment]
> Can we merge this @yyoncho ?

I guess we can. I will start working on migration to plists soon.

**Pull request merged**